_posts/2016-11-29-future.md

```diff
@@ -0,0 +1,239 @@
+
+[motiz88/babel-smoke-tests](https://github.com/motiz88/babel-smoke-tests) is a start!
+
+### API Documentation
+
+Docs should explain each feature and possible usages, improved links to books, articles, etc.
+
+- All plugins should have examples. Can also embed [RunKit](https://runkit.com/home) or the REPL.
+- Update Learn ES2015 to have ES2015+.
+- Add resources page
+- Update FAQ with common errors
+- Page for community plugins
+- babel-handbook, API docs, etc
+
+### Program Analysis
+
+- [Program Slicing](https://en.wikipedia.org/wiki/Program_slicing) with [slice-js](http://slides.com/kentcdodds/faster-javascript#/)
+- Semantic (AST aware) Grepping tool
+
+### `babel --debug`
+
+This command would print out all info (also when erroring). It would also include performance metrics on how long each plugin takes.
+
+### Parser Unity
+
+There have also been some discussions around parser/AST unity, in via [TheLarkInn/js-parser-discussions](https://github.com/TheLarkInn/js-parser-discussions) and previously with [estree](https://github.com/estree/estree).
+
+Unfortunately with Babel 6, we have "forked" and have a few differences in [our AST](https://github.com/babel/babylon#output) than ESTree. Babel inherently will want to support stage x features while other parsers may only want to support stage 4 features. We all might prioritize certain things more than others regarding spec compliancy, performance, stage x features, error messages, extensibility, releases, etc. However it's important for us to be open to breaking changes that may lead to better interop and a better community.
+
+### Sweet.js Interop
+
+Previous [issue](https://github.com/babel/babel/issues/568#issuecomment-71716260). Maybe we can just figure out how to have better interop instead?
+
+### Others?
+
+Anything not already brought up here? Please send us a tweet [@babeljs](https://twitter.com/babeljs), a message on our slack (join at [http://slack.babeljs.io/](slack.babeljs.io), or create an RFC-like issue in our repo!
+
+## How will it happen? Community
+
+Old Issues:
+
+- [Call for Contributors!](https://github.com/babel/babel/issues/1347)
+- [Being Prepared for Getting Hit by a Bus](https://github.com/babel/babel/issues/1888)
+
+You might think that as [a project gets more widely used](https://npm-stat.com/charts.html?package=babel-core) that more people show up to help out, but like most OSS projects that aren't backed by a company, there is a constant issue with maintenance and sustainability; people get burned out, move on to other cool projects, or get busy with work, life, family, etc.
+
+> TODO
+> // how to get involved, slack, mentoring, etc
```

Comment:
We could link to [easy contributions](https://github.com/babel/babel.github.io/issues?q=is%3Aopen+is%3Aissue+label%3Abeginner-friendly+label%3A%22help+wanted%22).